controllers/virtualmachine/volume/volume_controller.go

```diff
@@ -785,6 +793,110 @@ func (r *Reconciler) createCNSAttachment(
 	return nil
 }
 
+// handlePVCWithWFFC sets the selected-node annotation on an unbound PVC if it has
+// a WaitForFirstConsumer StorageClass.
+func (r *Reconciler) handlePVCWithWFFC(
+	ctx *pkgctx.VolumeContext,
+	volume vmopv1.VirtualMachineVolume,
+) error {
+
+	if volume.PersistentVolumeClaim == nil || volume.PersistentVolumeClaim.InstanceVolumeClaim != nil {
+		return nil
+	}
+
+	pvc := corev1.PersistentVolumeClaim{}
+	pvcKey := client.ObjectKey{
+		Namespace: ctx.VM.Namespace,
+		Name:      volume.PersistentVolumeClaim.ClaimName,
+	}
+
+	if err := r.Get(ctx, pvcKey, &pvc); err != nil {
+		return fmt.Errorf("cannot get PVC: %w", err)
+	}
+
+	if pvc.Status.Phase == corev1.ClaimBound {
+		// Regardless of the StorageClass binding mode there is nothing to done if already bound.
+		return nil
+	}
+
+	if pvc.Annotations[constants.KubernetesSelectedNodeAnnotationKey] != "" {
+		// Once set, this annotation cannot really be changed so just keep going.
+		// TBD: Make check if the Node still exists, and is in the VM's Zone?
+		return nil
+	}
+
+	scName := pvc.Spec.StorageClassName
+	if scName == nil || *scName == "" {
+		// While we have the DefaultStorageClass feature gate enabled, we don't mark an
+		// StorageClass with storageclass.kubernetes.io/is-default-class, so don't bother
+		// trying to resolve that here now and just continue the prior flow.
+		return nil
+	}
+
+	sc := &storagev1.StorageClass{}
+	if err := r.Get(ctx, client.ObjectKey{Name: *scName}, sc); err != nil {
+		return fmt.Errorf("cannot get StorageClass for PVC %s: %w", pvc.Name, err)
+	}
+
+	if mode := sc.VolumeBindingMode; mode == nil || *mode != storagev1.VolumeBindingWaitForFirstConsumer {
```

Comment:
Any value in using this: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-helpers/storage/volume/pv_helpers.go#L96?  Or do you think this will increase the dep graph significantly?

Reply:
Not really. We'd have to go implement another interface too